source/core/shape_parsing.cc

```diff
@@ -0,0 +1,351 @@
+                                            orientation);
+    }
+  else if (type == "cylinder")
+    {
+      if constexpr (dim == 3)
+        shape = std::make_shared<Cylinder<dim>>(shape_arguments[0],
+                                                shape_arguments[1],
+                                                position,
+                                                orientation);
+    }
+  else if (type == "cylindrical tube")
+    {
+      if constexpr (dim == 3)
+        shape = std::make_shared<CylindricalTube<dim>>(shape_arguments[0],
+                                                       shape_arguments[1],
+                                                       shape_arguments[2],
+                                                       position,
+                                                       orientation);
+    }
+  else if (type == "cylindrical helix")
+    {
+      if constexpr (dim == 3)
+        shape = std::make_shared<CylindricalHelix<dim>>(shape_arguments[0],
+                                                        shape_arguments[1],
+                                                        shape_arguments[2],
+                                                        shape_arguments[3],
+                                                        position,
+                                                        orientation);
+    }
+  else if (type == "cut hollow sphere")
+    {
+      if constexpr (dim == 3)
+        shape = std::make_shared<CutHollowSphere<dim>>(shape_arguments[0],
+                                                       shape_arguments[1],
+                                                       shape_arguments[2],
+                                                       position,
+                                                       orientation);
+    }
+  else if (type == "death star")
+    {
+      if constexpr (dim == 3)
+        shape = std::make_shared<DeathStar<dim>>(shape_arguments[0],
+                                                 shape_arguments[1],
+                                                 shape_arguments[2],
+                                                 position,
+                                                 orientation);
+    }
+  else if (type == "rbf")
+    {
+      shape =
+        std::make_shared<RBFShape<dim>>(shape_arguments, position, orientation);
+    }
+  else
+    StandardExceptions::ExcNotImplemented();
+  return shape;
+}
+
+template <int dim>
+std::shared_ptr<Shape<dim>>
+ShapeGenerator::initialize_shape_from_file(const std::string   type,
+                                           const std::string   file_name,
+                                           const Point<dim> &  position,
+                                           const Tensor<1, 3> &orientation)
+{
+  std::shared_ptr<Shape<dim>> shape;
+  std::vector<double>         shape_arguments;
+  if (type == "rbf")
+    {
+      if (file_name == "1") // Default case
```

Comment:
```suggestion
      if (file_name == "1") // Default case which is used for testing purposes
```
If it's only used for testing, it's not the regular case, but the exception no? Seems a bit off to put it first